sdk/core/Azure.Core/tests/public/JsonDataArrayTests.cs

```diff
@@ -234,14 +234,50 @@ public void CanSetObjectMemberViaArrayIndex()
         public void CanGetArrayLength()
         {
             dynamic data = JsonDataTestHelpers.CreateFromJson("[1, 2, 3]");
-            Assert.AreEqual(3, ((int[])data).Length);
+            Assert.AreEqual(3, data.Length);
+        }
+
+        [Test]
```

Comment:
We should consider adding tests for getting Length when the data is not an array and also not an array but exposes a `Length` property.

Reply:
Good idea - will add.